webapp/app/js/compareperf.js

```diff
@@ -214,25 +212,14 @@ comparePerf.controller('CompareCtrl', [ '$state', '$stateParams', '$scope', '$ro
     }
 
     function displayResults(rawResultsMap, newRawResultsMap) {
-      var counter = 0;
-      var compareResultsMap = {};
+      $scope.compareResults = {};
 
       $scope.testList.forEach(function(testName) {
-        if (counter > 0 && compareResultsMap[(counter-1)].headerColumns==2) {
-          counter--;
-        }
-
-        //TODO: figure out a cleaner method for making the names a header row
-        compareResultsMap[counter++] = {'name': testName.replace(' summary', ''),
-                                        'isEmpty': true, 'isMinor': false, 'headerColumns': 2,
-                                        'originalGeoMean': 'Old Rev', 'originalStddev': 'StdDev',
-                                        'newGeoMean': 'New Rev', 'newStddev': 'StdDev',
-                                        'delta': 'Delta', 'deltaPercentage': 'Delta'};
-
+        $scope.compareResults[testName] = [];
 
         $scope.platformList.forEach(function(platform) {
           var cmap = {'originalGeoMean': NaN, 'originalRuns': 0, 'originalStddev': NaN,
```

Comment:
While you're here, could you remove the quotes around the various properties? Would make it easier to read.